plugins/dominant-color-images/hooks.php

```diff
@@ -186,3 +186,82 @@ function dominant_color_render_generator(): void {
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_inline_style(): void {
+	$handle = 'dominant-color-admin-styles';
+	// PHPCS ignore reason: Version not used since this handle is only registered for adding an inline style.
+	// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = '.wp-core-ui .attachment-preview[data-dominant-color]:not(.has-transparency) { background-color: var(--dominant-color); }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_action( 'admin_enqueue_scripts', 'dominant_color_admin_inline_style' );
+
+/**
+ * Adds a script to the admin footer to modify the attachment template.
+ *
+ * This function injects a JavaScript snippet into the admin footer that modifies
+ * the attachment template. It adds attributes for dominant color and transparency
+ * to the template, allowing these properties to be displayed in the media library.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_admin_script(): void {
+	?>
+	<script type="module">
+		const tmpl = document.getElementById( 'tmpl-attachment' );
+		if ( tmpl ) {
+			tmpl.textContent = tmpl.textContent.replace( /^\s*<div[^>]*?(?=>)/, ( match ) => {
+				let replaced = match.replace( /\sclass="/, " class=\"{{ data.hasTransparency ? \'has-transparency\' : \'not-transparent\' }} " );
+				replaced += ' data-dominant-color="{{ data.dominantColor }}"';
+				replaced += ' data-has-transparency="{{ data.hasTransparency }}"';
+				replaced += ' style="--dominant-color: #{{ data.dominantColor }};"'; // TODO: Potentially there could be a style attribute as well!
+				return replaced;
+			} );
+		}
+	</script>
+	<?php
+}
+add_action( 'admin_print_footer_scripts', 'dominant_color_admin_script' );
+
+/**
+ * Prepares attachment data for JavaScript, adding dominant color and transparency information.
+ *
+ * This function enhances the attachment data for JavaScript by including information about
+ * the dominant color and transparency of the image. It modifies the response array to include
+ * these additional properties, which can be used in the media library interface.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<mixed>|mixed $response   The current response array for the attachment.
+ * @param WP_Post            $attachment The attachment post object.
+ * @param array<mixed>       $meta       The attachment metadata.
+ * @return array<mixed> The modified response array with added dominant color and transparency information.
+ */
+function dominant_color_prepare_attachment_for_js( $response, WP_Post $attachment, array $meta ): array {
+	if ( ! is_array( $response ) ) {
+		$response = array();
+	}
+
+	$response['dominantColor'] = '';
+	if ( isset( $meta['dominant_color'] ) && str_starts_with( '#', $meta['dominant_color'] ) ) {
+		$response['dominantColor'] = sanitize_hex_color( $meta['dominant_color'] );
+	}
+	$response['hasTransparency']      = '';
```

Comment:
This should fix the PHPCS complaint:

```suggestion
	$response['hasTransparency'] = '';
```